syntax = "proto2";
package Holoholo.Rpc;

message Inventory {
  optional bool success = 1;
  optional int32 ResponseNumber = 2;

  repeated GetInventoryOutProto Inventory = 100;
}

message GetInventoryOutProto {
  optional bool Success = 1;
  optional Holoholo.Rpc.InventoryDeltaProto InventoryDelta = 2;
 }

message InventoryDeltaProto {
  optional int64 OriginalTimestamp = 1;
  optional int64 NewTimestamp = 2;
  repeated InventoryItemProto InventoryItem = 3;
 }

 message InventoryItemProto {
  optional int64 ModifiedTimestamp = 1;
  optional int64 DeletedItemKey = 2;
  optional HoloInventoryItemProto Item = 3;
 }

message HoloInventoryItemProto {
  optional Holoholo.Rpc.PokemonProto Pokemon = 1;
  optional Holoholo.Rpc.ItemProto Item = 2;
  optional Holoholo.Rpc.PokedexEntryProto PokedexEntry = 3;
  optional Holoholo.Rpc.PlayerStatsProto PlayerStats = 4;
  optional Holoholo.Rpc.PlayerCurrencyProto PlayerCurrency = 5;
  optional Holoholo.Rpc.PlayerCameraProto PlayerCamera = 6;
  optional Holoholo.Rpc.InventoryUpgradesProto InventoryUpgrades = 7;
  optional Holoholo.Rpc.AppliedItemsProto AppliedItems = 8;
  optional Holoholo.Rpc.EggIncubatorsProto EggIncubators = 9;
  optional Holoholo.Rpc.PokemonFamilyProto PokemonFamily = 10;
 }

message PokemonProto {
  optional uint64 Id = 1;
  optional int32 PokemonId = 2;
  optional int32 Cp = 3;
  optional int32 Stamina = 4;
  optional int32 MaxStamina = 5;
  optional int32 Move1 = 6;
  optional int32 Move2 = 7;
  optional string DeployedFortId = 8;
  optional string OwnerName = 9;
  optional bool IsEgg = 10;
  optional double EggKmWalkedTarget = 11;
  optional double EggKmWalkedStart = 12;
  // ENUM.Holoholo.Rpc.PokemonCreateContext Origin = 14;
  optional int32 Origin = 14;
  optional float HeightM = 15;
  optional float WeightKg = 16;
  optional int32 IndividualAttack = 17;
  optional int32 IndividualDefense = 18;
  optional int32 IndividualStamina = 19;
  optional float CpMultiplier = 20;
  // ENUM.Holoholo.Rpc.Item Pokeball = 21;
  optional int32 Pokeball = 21;
  optional int64 CapturedS2CellId = 22;
  optional int32 BattlesAttacked = 23;
  optional int32 BattlesDefended = 24;
  optional string EggIncubatorId = 25;
  optional int64 CreationTimeMs = 26;
  optional int32 NumUpgrades = 27;
  optional float AdditionalCpMultiplier = 28;
  optional bool Favorite = 29;
  optional string Nickname = 30;
  optional bool FromFort = 31;
}

message ItemProto {
  // ENUM.Holoholo.Rpc.Item Item = 1;
  optional int32 Item = 1;
  optional int32 Count = 2;
  optional bool Unseen = 3;
 }

 message PokedexEntryProto {
  optional int32 PokedexEntryNumber = 1;
  optional int32 TimesEncountered = 2;
  optional int32 TimesCaptured = 3;
  optional int32 EvolutionStonePieces = 4;
  optional int32 EvolutionStones = 5;
 }

  message PlayerStatsProto {
  optional int32 Level = 1;
  optional int64 Experience = 2;
  optional int64 PrevLevelExp = 3;
  optional int64 NextLevelExp = 4;
  optional float KmWalked = 5;
  optional int32 NumPokemonEncountered = 6;
  optional int32 NumUniquePokedexEntries = 7;
  optional int32 NumPokemonCaptured = 8;
  optional int32 NumEvolutions = 9;
  optional int32 PokeStopVisits = 10;
  optional int32 NumberOfPokeballThrown = 11;
  optional int32 NumEggsHatched = 12;
  optional int32 BigMagikarpCaught = 13;
  optional int32 NumBattleAttackWon = 14;
  optional int32 NumBattleAttackTotal = 15;
  optional int32 NumBattleDefendedWon = 16;
  optional int32 NumBattleTrainingWon = 17;
  optional int32 NumBattleTrainingTotal = 18;
  optional int32 PrestigeRaisedTotal = 19;
  optional int32 PrestigeDroppedTotal = 20;
  optional int32 NumPokemonDeployed = 21;
  //unknown NumPokemonCaughtByType = 22;
  optional int32 NumPokemonCaughtByType = 22;
  optional int32 SmallRattataCaught = 23;
 }

 message PlayerCurrencyProto {
  optional int32 Gems = 1;
 }
 message PlayerCameraProto {
  optional bool DefaultCamera = 1;
 }
 message InventoryUpgradesProto {
  //unknown InventoryUpgrade = 1;
 }
 message InventoryUpgradeProto {
  // ENUM.Holoholo.Rpc.Item Item = 1;
  optional int32 Item = 1;
  // ENUM.Holoholo.Rpc.InventoryUpgradeType UpgradeType = 2;
  optional int32 UpgradeType = 2;
  optional int32 AdditionalStorage = 3;
 }
 message EggIncubatorsProto {
  //unknown EggIncubator = 1;
 }
 message EggIncubatorProto {
  optional string ItemId = 1;
  // ENUM.Holoholo.Rpc.Item Item = 2;
  optional int32 Item = 2;
  // ENUM.Holoholo.Rpc.EggIncubatorType IncubatorType = 3;
  optional int32 IncubatorType = 3;
  optional int32 UsesRemaining = 4;
  optional int64 PokemonId = 5;
  optional double StartKmWalked = 6;
  optional double TargetKmWalked = 7;
 }
 message AppliedItemsProto {
  //unknown Item = 4;
 }
 message AppliedItemProto {
  // ENUM.Holoholo.Rpc.Item Item = 1;
  optional int32 Item = 1;
  // ENUM.Holoholo.Rpc.HoloItemType ItemType = 2;
  optional int32 ItemType = 2;
  optional int64 ExpirationMs = 3;
  optional int64 AppliedMs = 4;
 }
 message PokemonFamilyProto {
  // ENUM.Holoholo.Rpc.HoloPokemonFamilyId FamilyId = 1;
  optional int32 FamilyId = 1;
  optional int32 Candy = 2;
 }

